samples/cellular/nrf_provisioning/README.rst

```diff
@@ -0,0 +1,137 @@
+.. _nrf_provisioning_sample:
+
+cellular: nRF Device provisioning
+#################################
+
+.. contents::
+   :local:
+   :depth: 2
+
+This sample demonstrates how to use the :ref:`lib_nrf_provisioning` service on your device.
+
+Requirements
+************
+
+The sample supports the following development kit:
+
+.. table-from-sample-yaml::
+
+.. include:: /includes/tfm.txt
+
+The sample requires that private device key is installed on the device and the associated Device UUID is obtained from the Identity Service.
+
+.. note::
+   This sample requires modem firmware v2.0.0 or later.
+
+Overview
+********
+
+The sample shows how the device performs the following actions:
+
+* Connects to nRF Device provisioning Service.
+* Fetches available device specific provisioning commands.
+* Decodes the commands.
+* Acts on any AT commands, if available.
+* Reports the results back to the server.
+  In case of errors, stops processing the commands at the first error and reports it back to server.
+* Sends ``FINISHED`` response if all the previous commands are executed without errors provided and ``FINISHED`` is one of the set provisioning commands.
+
+User interface
+**************
+
+Device side interaction is not required.
+You must fill the command queue at the server side.
+
+Configuration
+*************
+
+|config|
+
+Configuration options
+=====================
+
+Check and configure the following configuration options for the sample:
+
+.. _CONFIG_NRF_PROVISIONING_ROOT_CA_SEC_TAG:
+
+CONFIG_NRF_PROVISIONING_ROOT_CA_SEC_TAG
+   Root CA security tag for the Nordic Semiconductor's provisioning service.
+   Needs to be set explicitly and if not, the compilation fails.
+
+.. _CONFIG_NRF_PROVISIONING_HTTP_HOSTNAME:
+
+CONFIG_NRF_PROVISIONING_HTTP_HOSTNAME
+   Hostname of the nRF Device provisioning service
+
+.. _CONFIG_NRF_PROVISIONING_HTTP_PORT:
+
+CONFIG_NRF_PROVISIONING_HTTP_PORT
+   HTTP port of the nRF Device provisioning service
+
+.. _CONFIG_NRF_PROVISIONING_HTTP_TIMEOUT_MS:
+
+CONFIG_NRF_PROVISIONING_HTTP_TIMEOUT_MS
+   HTTP timeout
+
+.. _CONFIG_RF_PROVISIONING_HTTP_RX_BUF_S:
+
+CONFIG_RF_PROVISIONING_HTTP_RX_BUF_S
+   Response payload buffer size
+
+.. _CONFIG_NRF_PROVISIONING_HTTP_TX_BUF_SZ:
+
+CONFIG_NRF_PROVISIONING_HTTP_TX_BUF_SZ
+   Command request body size
+
+Building and running
+********************
+
+.. |sample path| replace:: :file:`samples/cellular/nrf_provisioning`
+
+.. include:: /includes/build_and_run_ns.txt
+
+Testing
+=======
+
+|test_sample|
+
+#. |connect_kit|
+#. |connect_terminal|
+#. Add provisioning command(s) to nRF Provisioning Service.
+#. Power on or reset your device.
+#. Observe that the sample starts and connects to the LTE network.
+#. Observe that provisioning pauses and resumes while fetching and executing provisioning commands.
+
+Sample Output
+=============
+
+The following is an example output of the sample when there is no commands queued on server side:
```

Comment:
```suggestion
The following is an example output of the sample when there are no commands queued on server side:
```
Subject-verb agreement, but if this refers to provisioning commands in the Provisioning Service, please refer to it as *provisioning configuration*.